index.bs

```diff
@@ -4937,6 +5027,50 @@ side=], or to terminate or error the stream.
  1. Return [=this=].\[[writable]].
 </div>
 
+<h4 id="ts-transfer">Transfer via `postMessage()`</h4>
+
+<dl class="domintro">
+ <dt><code>destination.postMessage(ts, { transfer: [ts] });</code>
+ <dd>
+  <p>Sends a {{TransformStream}} to another frame, window, or worker.
+
+  <p>The transferred stream can be used exactly like the original. Its [=readable side|readable=]
+  and [=writable sides=] will become locked and no longer directly usable.
+ </dd>
+</dl>
+
+<div algorithm="TransformStream transfer steps">
+ {{TransformStream}} objects are [=transferable objects=]. Their [=transfer steps=], given |value|
+ and |dataHolder|, are:
+
+ 1. Let |readable| be |value|.\[[readable]].
+ 1. Let |writable| be |value|.\[[writable]].
+ 1. If ! [$IsReadableStreamLocked$](|readable|) is true, throw a "{{DataCloneError}}"
+    {{DOMException}}.
+ 1. If ! [$IsWritableStreamLocked$](|writable|) is true, throw a "{{DataCloneError}}"
+    {{DOMException}}.
```

Comment:
Adding these checks here, as opposed to delegating to transferring the readable and writable sides independently, avoids ending up with a "half-transferred" transform stream. That's nice :). I wonder if we should `<p class="note">` that? In either case we should be sure we have a test for it.

Reply:
I've updated the tests to test this feature (which is missing from Blink's implementation 😳).

I don't feel I need to call attention to the feature with a note, as it is just a matter of polish.